packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts

```diff
@@ -35,45 +35,50 @@ describe.skip("S3 File Plugin", () => {
       }
     )
   })
+  ;(["public", "private"] as const).forEach((access) => {
+    it("uploads, reads, and then deletes a file successfully", async () => {
+      const fileContent = await fs.readFile(fixtureImagePath)
+      const fixtureAsBinary = fileContent.toString("binary")
 
-  it("uploads, reads, and then deletes a file successfully", async () => {
-    const fileContent = await fs.readFile(fixtureImagePath)
-    const fixtureAsBinary = fileContent.toString("binary")
+      const resp = await s3Service.upload({
+        filename: "catphoto.jpg",
+        mimeType: "image/jpeg",
+        content: fixtureAsBinary,
+        access,
+      })
 
-    const resp = await s3Service.upload({
-      filename: "catphoto.jpg",
-      mimeType: "image/jpeg",
-      content: fixtureAsBinary,
-    })
+      expect(resp).toEqual({
+        key: expect.stringMatching(/tests\/catphoto.*\.jpg/),
+        url: expect.stringMatching(/https:\/\/.*\.jpg/),
+      })
 
-    expect(resp).toEqual({
-      key: expect.stringMatching(/tests\/catphoto.*\.jpg/),
-      url: expect.stringMatching(/https:\/\/.*\.jpg/),
-    })
+      const urlResp = await axios.get(resp.url).catch((e) => e.response)
+      expect(urlResp.status).toEqual(access === "public" ? 200 : 403)
 
-    const signedUrl = await s3Service.getPresignedDownloadUrl({
-      fileKey: resp.key,
-    })
+      const signedUrl = await s3Service.getPresignedDownloadUrl({
+        fileKey: resp.key,
+      })
 
-    const signedUrlFile = Buffer.from(
-      await axios
-        .get(signedUrl, { responseType: "arraybuffer" })
-        .then((r) => r.data)
-    )
+      const signedUrlFile = Buffer.from(
+        await axios
+          .get(signedUrl, { responseType: "arraybuffer" })
+          .then((r) => r.data)
+      )
 
-    expect(signedUrlFile.toString("binary")).toEqual(fixtureAsBinary)
+      expect(signedUrlFile.toString("binary")).toEqual(fixtureAsBinary)
 
-    await s3Service.delete({ fileKey: resp.key })
+      await s3Service.delete({ fileKey: resp.key })
 
-    // TODO: Currently the presignedURL will be returned even if the file doesn't exist. Should we check for existence first?
-    const deletedFileUrl = await s3Service.getPresignedDownloadUrl({
-      fileKey: resp.key,
-    })
+      // TODO: Currently the presignedURL will be returned even if the file doesn't exist. Should we check for existence first?
```

Comment:
It sounds reasonable to check for existence in `getPresignedDownloadUrl` and throw `File not found` or similar, if that is the case.

Reply:
@olivermrbl Yeah I think it makes sense, the only downside is that it requires one more request, so I was a bit unsure if we want to make the 99% of the requests slower for better error message.

Now that I think about it though, presigned URLs are only valid for private files, which will be a lot less common, so maybe we can add the check. WDYT?

Btw I'll just merge this PR as it is unrelated, and depending on what we decide I'll open a new one.

Reply:
Yeah, sounds good to me. No need to rush with the follow-up PR though. Can be tackled later.

Reply:
I'll just open a ticket for now